components/Modal.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import IconTwitter from '../components/IconTwitter';
+
+type Props = {
+  children: React.ReactNode;
+  toggleShow: boolean;
+  setToggleShow: (toggleShow: boolean) => void;
+};
+
+function Modal({ children, toggleShow, setToggleShow }: Props) {
+  return (
+    <>
+      {toggleShow && (
```

Comment:
You should make the conditional rendering where the Modal component is used (in index.tsx aparently)